cvmfs/loader_talk.cc

```diff
@@ -66,22 +67,38 @@ static void *MainTalk(void *data __attribute__((unused))) {
     char command;
     ReloadMode reload_mode = kReloadLegacy;
     if (recv(con_fd, &command, 1, 0) > 0) {
-      if ((command == 'd') || (command == 'n')) {
-        // version that specifies reloading in debug or non-debug mode
-        // receives 2 commands
-        // first: debug (d) / non-debug(n)
-        // second: 'R' or 'S'
-        reload_mode = command == 'd' ? kReloadDebug : kReloadNoDebug;
-        if (recv(con_fd, &command, 1, 0) > 0) {
-          if ((command != 'R') && (command != 'S')) {
-            SendMsg2Socket(con_fd, "unknown command\n");
-            continue;
-          }
+      bool unkown_command = true;
+      if ((command == 'S') || (command == 'R')) {
+        struct pollfd fd;
+        int ret;
+
+        // check if second command arrives (in legacy no 2nd cmd arrives)
+        fd.fd = con_fd;
+        fd.events = POLLIN;
+        ret = poll(&fd, 1, 1000);  // 1 sec timeout
+        switch (ret) {
+          case -1:
+            // Error
+          break;
+          case 0:
+            // Timeout = Legacy Version (cannot switch debug on/off)
+            unkown_command = false;
+          break;
+          default:  // Version that can set debug on/off
+            char second_cmd;
+            if (recv(con_fd, &second_cmd, 1, 0) > 0) {
+              if ((second_cmd == 'd') || (second_cmd == 'n')) {
+                reload_mode = second_cmd == 'd' ? kReloadDebug : kReloadNoDebug;
+                unkown_command = false;
+              }
+            }
+          break;
         }
-      } else if ((command != 'R') && (command != 'S')) {  // legacy support
+      }
+      if (unkown_command) {
         SendMsg2Socket(con_fd, "unknown command\n");
```

Comment:
printing the unknown command char could be helpful

Reply:
it is printed thats what `sendmsg2socket` does. (you might remember the error message wie had before `<repo>.cern.ch: unknown command` ... REPO CRASHED etc. that came from there